signer/src/block_observer.rs

```diff
@@ -369,24 +395,20 @@ mod tests {
             storage.clone(),
             test_harness.clone(),
         );
+        let _signal_rx = ctx.get_signal_receiver();
```

Comment:
Oh, this is needed to make sure that all receivers aren't dropped. Maybe add a comment here?

Reply:
Yeah exactly, haha I had considered it teetered towards not 😅 Yeah I'll add one

Reply:
Updated in [e7064d1](https://github.com/stacks-network/sbtc/pull/580/commits/e7064d1c891fc1e5d30906dc33ba69bed4970f6e)